pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,831 @@
+use super::*;
+use drink::{
+	sandbox_api::assets_api::AssetsAPI,
+	session::{ContractBundle, Session},
+};
+use frame_support::assert_ok;
+use pop_api::{
+	primitives::{
+		ArithmeticError::Overflow,
+		Error::{Arithmetic, Module},
+		TokenId,
+	},
+	v0::fungibles::events::{Approval, Created, Transfer},
+};
+use pop_sandbox::{Balance, DevnetSandbox as Sandbox, ALICE, BOB};
+use scale::Encode;
+use std::sync::LazyLock;
+use utils::*;
+
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+const AMOUNT: Balance = MIN_BALANCE * 4;
+const MIN_BALANCE: Balance = 10_000;
+const TOKEN: TokenId = 1;
+
+static CONTRACT: LazyLock<ContractBundle> = LazyLock::new(|| BundleProvider::local().unwrap());
+
+/// Deployment and constructor method tests.
+
+#[drink::test(sandbox = Sandbox)]
+fn new_constructor_works(mut session: Session) {
+	let _ = env_logger::try_init();
+	// Fails to deploy contract with an used token ID.
+	let token = TOKEN + 1;
+	assert_ok!(session.sandbox().create(&token, &ALICE, MIN_BALANCE));
+	// `pallet-assets` returns `InUse` error.
+	assert_eq!(
+		deploy(
+			&mut session,
+			CONTRACT.clone(),
+			"new",
+			vec![token.to_string(), MIN_BALANCE.to_string()]
+		),
+		Err(into_psp22_custom(Module { index: 52, error: [5, 0] }))
```

Comment:
This isnt great. Why should the dev need to know about this strange error type (Module { .. })? If I saw this as a potential user I would be put off. We are supposed to be making things easier, showing how simple it can be, but then use somewhat advanced things which the dev shouldnt need to worry about.

Reply:
The primary purpose of this test code is to demonstrate the effectiveness of pop_api errors. This will become even more apparent once the changes proposed in [this PR](https://github.com/r0gue-io/pop-node/pull/300/files/bea0e9aa4b9f9e231b2be5ad25f974bf290913fc#r1778358651) are implemented. That's why you'll notice that each test case includes at least one scenario involving a Pop API error from `StatusCode`, rather than a `PSP22Error`.

cc: @Daanvdplas

Reply:
Resolved in the PR: https://github.com/r0gue-io/pop-node/pull/324